niworkflows/anat/mni.py

```diff
@@ -138,8 +156,43 @@ def _config_ants(self, ants_settings):
         if self.inputs.testing:
             resolution = 2
 
-        norm.inputs.fixed_image = op.join(
+        self.norm.inputs.fixed_image = op.join(
             mni_template, '%dmm_%s.nii.gz' % (resolution, self.inputs.reference))
-        norm.inputs.fixed_image_mask = op.join(mni_template,
-                                               '%dmm_brainmask.nii.gz' % resolution)
-        return norm
+        self.norm.inputs.fixed_image_mask = op.join(
+            mni_template, '%dmm_brainmask.nii.gz' % resolution)
+
+
+    def _generate_report(self):
+        """ Generates the visual report """
+        NIWORKFLOWS_LOG.debug('Generating visual report')
+
+        def _plot_xyz(anat_img, plot_params, div_id):
+            """
+
+            """
+            NIWORKFLOWS_LOG.info('Plotting %s', str(anat_img))
+            svgs = []
+
+            for mode, cuts in list(DEFAULT_CUTS.items()):
+                plot_params['display_mode'] = mode
+                plot_params['cut_coords'] = cuts
+                image = plot_anat(anat_img, **plot_params)
+                svgs.append('<div id="{0}-{1}" class="{0}">{2}</div>'.format(
+                    div_id, mode, report.as_svg(image)))
+                image.close()
+            return '\n'.join(svgs)
+
+        plot_params = {'cut_coords': 3}
+        report.save_html(
+            template='overlay_3d_report.tpl',
+            report_file_name=self._results['html_report'],
+            unique_string='antsRegistration',
```

Comment:
unique_string: Yeah, I was reluctant to add another dependency to the code. I would prefer to pick up the identifier of the nipype interface or so.

div_id for plots: this is an attempt to be able to easily select subplots with css. When you integrate the code in a larger html file, things don't work as expected (not your fault). In the long term, my idea was (still is) to have only one svg file, that composes the 3 panels of the background and the 3 panels of the foreground. Since svg is html compatible, the css tricks work inside this unique svg with the 6 panels.

So this divs are mocking the inner svg elements of that future panel.

Does it make sense?